lapce-proxy/src/plugin.rs

```diff
@@ -183,6 +187,65 @@ impl PluginCatalog {
         Ok(plugin)
     }
 
+    fn get_plugin_env(
+        &mut self,
+        plugin_desc: &PluginDescription,
+    ) -> Vec<(String, String)> {
+        let mut vars = Vec::new();
+
+        let conf = match plugin_desc.configuration.clone() {
```

Comment:
I think we could avoid this `clone` here. Instead of `serde_json::from_value` you can deserialize a type like `PluginConfiguration::deserialize(conf)` and I *think* that can take a reference.

Reply:
Nice catch! this is very interesting to see as a learning example for me, thanks!